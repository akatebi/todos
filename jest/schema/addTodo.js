const { fetch } = require("../fetch");

export const addTodo = ({ text, userId, clientMutationId }) => {
  const variables = { text, userId, clientMutationId };
  const query = `mutation addTodo($text: String!, $userId: ID!, $clientMutationId: String) {
        addTodo(
          input: { text: $text, userId: $userId, clientMutationId: $clientMutationId }
        ) {
          user {
            email
            totalCount
            completedCount
          }
          todoEdge {
            cursor
            node {
              id
              text
              complete
            }
          }
          clientMutationId
        }
      }`;
  return fetch({ query, variables });
};

