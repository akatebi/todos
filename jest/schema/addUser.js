const { fetch } = require("../fetch");

export const addUser = ({ email, clientMutationId }) => {
  const variables = { email, clientMutationId };
  const query = `mutation AddUser($email: String!, $clientMutationId: String ) {
    addUser(input: { email: $email, clientMutationId: $clientMutationId }) {
      id
      clientMutationId
    }}`;
  return fetch({ query, variables });
};


