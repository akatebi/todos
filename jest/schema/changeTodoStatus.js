const { fetch } = require("../fetch");

const ChangeTodoStatus = ({ complete, id, userId, clientMutationId }) => {
  const variables = { complete, id, userId, clientMutationId };
  const query = `mutation ChangeTodoStatus(
    $complete: Boolean!
    $id: ID!
    $userId: ID!
    $clientMutationId: String
  ) {
    changeTodoStatus(input: { 
      complete: $complete, id: $id, userId: $userId, 
      clientMutationId: $clientMutationId }) {
      todo {
        id
        text
        complete
      }
      user {
        id
        email
        totalCount
        completedCount
      }
      clientMutationId
    }
  }  
  `;
  return fetch({ query, variables });
};

const ChangeTodoStatusTest = (complete) => () => {
  it("", async () => {
    const userId = global.userId;
    for (let i = 0; i < 3; i++) {
      const id = todoIds[i];
      const clientMutationId = `ChangeTodoStatus-${i}`;
      const resp = await ChangeTodoStatus({
        complete,
        id,
        userId,
        clientMutationId,
      });
      expect(resp).toMatchSnapshot();
    }
  });
};

export default ChangeTodoStatusTest;
