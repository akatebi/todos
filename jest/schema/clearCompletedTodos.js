const { fetch } = require("../fetch");

export const clearCompletedTodos = ({ userId, clientMutationId }) => {
    const variables = { userId, clientMutationId };
    const query = `mutation clearCompletedTodos($userId: ID!, $clientMutationId: String) {
      clearCompletedTodos(input: {userId: $userId, clientMutationId: $clientMutationId}) {
        deletedTodoIds
        user {
          id
          completedCount
          totalCount
        }
        clientMutationId
      }
    }
    `;
    return fetch({ query, variables });
  };
  
  