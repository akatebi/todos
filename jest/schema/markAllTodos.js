const { fetch } = require("../fetch");

const MarkAllTodos = ({ complete, userId, clientMutationId }) => {
    const variables = { complete, userId, clientMutationId };
    const query = `mutation markAllTodos($complete: Boolean!, $userId: ID!, $clientMutationId: String) {
      markAllTodos(input: { complete: $complete, userId: $userId, clientMutationId: $clientMutationId}) {
        user {
          id
          completedCount
          totalCount
        }
        changedTodos {
          id
          text
          complete
        }
        clientMutationId
      }
    }
    `;
    return fetch({ query, variables });
  };
  
  const MarkAllTodosTest = () => () => {
    it("", async () => {
      const userId = global.userId;
      const clientMutationId = "MarkAllTodos";
      const complete = true;
      const resp = await MarkAllTodos({ complete, userId, clientMutationId });
      expect(resp).toMatchSnapshot();
    });
  };
  
export default MarkAllTodosTest;