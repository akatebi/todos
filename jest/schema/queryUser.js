const { fetch } = require("../fetch");

export const queryUser = ({ email, first = 100 }) => {
  const variables = { email };
  const query = `query User($email: String!, $status: Status, $first: Int, $after: String) {
        user(email: $email) {
          id
          email
          completedCount
          totalCount
          todos(status: $status, first: $first, after: $after) {
            edges {
              cursor
              node {
                id
                complete
                __typename
              }
            }
            pageInfo {
              endCursor
              hasNextPage
            }
          }
        }
      }`;
  return fetch({ query, variables });
};

