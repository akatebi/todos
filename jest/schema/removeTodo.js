const { fetch } = require("../fetch");

export const removeTodo = ({ id, clientMutationId }) => {
  const variables = { id, clientMutationId };
  const query = `mutation removeTodo($id: ID!, $clientMutationId: String) {
        removeTodo(
          input: { id: $id, clientMutationId: $clientMutationId }
        ) {
          user {
            id
            email
            totalCount
            completedCount
          }
          clientMutationId
        }
      }`;
  return fetch({ query, variables });
};

