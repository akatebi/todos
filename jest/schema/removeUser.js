const { fetch } = require("../fetch");

export const removeUser = ({ email, clientMutationId }) => {
  const variables = { email, clientMutationId };
  const query = `mutation RemoveUser($email: String!, $clientMutationId: String) {
    removeUser(input: { email: $email, clientMutationId: $clientMutationId }) {
      clientMutationId
    }}`;
  return fetch({ query, variables });
};

