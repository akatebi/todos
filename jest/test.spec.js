import { removeUser } from "./schema/removeUser";
import { addUser } from "./schema/addUser";
import { addTodo } from "./schema/addTodo";
import { renameTodo } from "./schema/renameTodo";
import { removeTodo } from "./schema/removeTodo";
import { queryUser } from "./schema/queryUser";
import { markAllTodos } from "./schema/markAllTodos";
import { clearCompletedTodos } from "./schema/clearCompletedTodos";
import { changeTodoStatus } from "./schema/changeTodoStatus";

describe("Todos GraphQL", () => {
    test("Query & Mutation", async() => {
        const count = 1;
        let clientMutationId, email, resp;
        clientMutationId = "1";
        email = "user@test.com";
        resp = await removeUser({ email, clientMutationId });
        expect("removeUser").toMatchSnapshot();
        expect(resp).toMatchSnapshot();
        
        clientMutationId = "2";
        resp = await addUser({ email, clientMutationId });
        expect(resp).toMatchSnapshot();
        const userId = resp.data.addUser.id;

        const todoIds = [];
        clientMutationId = "3";
        for (let i = 0; i < count; i++) {
            const text = `Get A Customer ${i + 1}`;
            const cid = `${clientMutationId}.${i}`;
            resp = await addTodo({ text, userId, clientMutationId: cid });
            expect(resp.data.addTodo.user.totalCount).toBe(i+1)
            expect(resp).toMatchSnapshot();
            todoIds.push(resp.data.addTodo.todoEdge.node.id);
        }
        
        resp = await queryUser({ email });
        expect(resp).toMatchSnapshot();

        clientMutationId = "4";
        let complete = true;
        resp = await markAllTodos({ complete, userId, clientMutationId });
        expect(resp.data.markAllTodos.user.totalCount).toBe(count)
        expect(resp.data.markAllTodos.user.completedCount).toBe(count)
        expect(resp).toMatchSnapshot();
        
        complete = false;
        clientMutationId = "5";
        for (let i = 0; i < count; i++) {
            const id = todoIds[i];
            const cid = `${clientMutationId}.${i}`;
            const resp = await changeTodoStatus({
                complete, id, userId, clientMutationId: cid
            });
            expect(resp.data.changeTodoStatus.user.completedCount).toBe(count-(i+1))
            expect(resp).toMatchSnapshot();
        }
        
        complete = true;
        clientMutationId = "6";
        for (let i = 0; i < count; i++) {
            const id = todoIds[i];
            const cid = `${clientMutationId}.${i}`;
            const resp = await changeTodoStatus({ complete, id, userId, clientMutationId: cid });
            expect(resp.data.changeTodoStatus.user.completedCount).toBe(i+1)
            expect(resp).toMatchSnapshot();
        }

        clientMutationId = "7";
        resp = await clearCompletedTodos({ userId, clientMutationId });
        expect(resp.data.clearCompletedTodos.user.completedCount).toBe(0)
        expect(resp).toMatchSnapshot();

        clientMutationId = "8";
        const text = "Get A Job"
        await addTodo({ text, userId, clientMutationId })
        
        clientMutationId = "9";
        for (let i = 0; i < count; i++) {
            const id = todoIds[i];
            const text = `renamed ${i}`;
            const cid = `${clientMutationId}.${i}`;
            const resp = await renameTodo({ id, text, clientMutationId: cid });
            expect(resp).toMatchSnapshot();
        }

        clientMutationId = "10";
        for (let i = 0; i < count; i++) {
            const id = todoIds[i];
            const cid = `${clientMutationId}.${i}`;
            const resp = await removeTodo({ id, userId, clientMutationId: cid });
            expect(resp).toMatchSnapshot();
        }

        resp = await queryUser({ email });
        expect(resp).toMatchSnapshot();
        
    });
});
