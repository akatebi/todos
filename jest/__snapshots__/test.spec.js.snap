// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos GraphQL Query & Mutation 1`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 2`] = `
Object {
  "query": "mutation RemoveUser($email: String!, $clientMutationId: String) {
    removeUser(input: { email: $email, clientMutationId: $clientMutationId }) {
      clientMutationId
    }}",
  "variables": Object {
    "clientMutationId": "1",
    "email": "user@test.com",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 3`] = `"removeUser"`;

exports[`Todos GraphQL Query & Mutation 4`] = `
Object {
  "data": Object {
    "removeUser": Object {
      "clientMutationId": "1",
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 5`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 6`] = `
Object {
  "query": "mutation AddUser($email: String!, $clientMutationId: String ) {
    addUser(input: { email: $email, clientMutationId: $clientMutationId }) {
      id
      clientMutationId
    }}",
  "variables": Object {
    "clientMutationId": "2",
    "email": "user@test.com",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 7`] = `
Object {
  "data": Object {
    "addUser": Object {
      "clientMutationId": "2",
      "id": "VXNlcjo5MA==",
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 8`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 9`] = `
Object {
  "query": "mutation addTodo($text: String!, $userId: ID!, $clientMutationId: String) {
        addTodo(
          input: { text: $text, userId: $userId, clientMutationId: $clientMutationId }
        ) {
          user {
            email
            totalCount
            completedCount
          }
          todoEdge {
            cursor
            node {
              id
              text
              complete
            }
          }
          clientMutationId
        }
      }",
  "variables": Object {
    "clientMutationId": "3.0",
    "text": "Get A Customer 1",
    "userId": "VXNlcjo5MA==",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 10`] = `
Object {
  "data": Object {
    "addTodo": Object {
      "clientMutationId": "3.0",
      "todoEdge": Object {
        "cursor": "MjI1",
        "node": Object {
          "complete": false,
          "id": "VG9kbzoyMjU=",
          "text": "Get A Customer 1",
        },
      },
      "user": Object {
        "completedCount": 0,
        "email": "user@test.com",
        "totalCount": 1,
      },
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 11`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 12`] = `
Object {
  "query": "query User($email: String!, $status: Status, $first: Int, $after: String) {
        user(email: $email) {
          id
          email
          completedCount
          totalCount
          todos(status: $status, first: $first, after: $after) {
            edges {
              cursor
              node {
                id
                complete
                __typename
              }
            }
            pageInfo {
              endCursor
              hasNextPage
            }
          }
        }
      }",
  "variables": Object {
    "email": "user@test.com",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 13`] = `
Object {
  "data": Object {
    "user": Object {
      "completedCount": 0,
      "email": "user@test.com",
      "id": "VXNlcjo5MA==",
      "todos": Object {
        "edges": null,
        "pageInfo": Object {
          "endCursor": null,
          "hasNextPage": true,
        },
      },
      "totalCount": 1,
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 14`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 15`] = `
Object {
  "query": "mutation markAllTodos($complete: Boolean!, $userId: ID!, $clientMutationId: String) {
      markAllTodos(input: { complete: $complete, userId: $userId, clientMutationId: $clientMutationId}) {
        user {
          id
          completedCount
          totalCount
        }
        changedTodos {
          id
          text
          complete
        }
        clientMutationId
      }
    }
    ",
  "variables": Object {
    "clientMutationId": "4",
    "complete": true,
    "userId": "VXNlcjo5MA==",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 16`] = `
Object {
  "data": Object {
    "markAllTodos": Object {
      "changedTodos": Array [
        Object {
          "complete": true,
          "id": "VG9kbzoyMjU=",
          "text": "Get A Customer 1",
        },
      ],
      "clientMutationId": "4",
      "user": Object {
        "completedCount": 1,
        "id": "VXNlcjo5MA==",
        "totalCount": 1,
      },
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 17`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 18`] = `
Object {
  "query": "mutation ChangeTodoStatus(
    $complete: Boolean!
    $id: ID!
    $userId: ID!
    $clientMutationId: String
  ) {
    changeTodoStatus(input: { 
      complete: $complete, id: $id, userId: $userId, 
      clientMutationId: $clientMutationId }) {
      todo {
        id
        text
        complete
      }
      user {
        id
        email
        totalCount
        completedCount
      }
      clientMutationId
    }
  }  
  ",
  "variables": Object {
    "clientMutationId": "5.0",
    "complete": false,
    "id": "VG9kbzoyMjU=",
    "userId": "VXNlcjo5MA==",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 19`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "5.0",
      "todo": Object {
        "complete": false,
        "id": "VG9kbzoyMjU=",
        "text": "Get A Customer 1",
      },
      "user": Object {
        "completedCount": 0,
        "email": "user@test.com",
        "id": "VXNlcjo5MA==",
        "totalCount": 1,
      },
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 20`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 21`] = `
Object {
  "query": "mutation ChangeTodoStatus(
    $complete: Boolean!
    $id: ID!
    $userId: ID!
    $clientMutationId: String
  ) {
    changeTodoStatus(input: { 
      complete: $complete, id: $id, userId: $userId, 
      clientMutationId: $clientMutationId }) {
      todo {
        id
        text
        complete
      }
      user {
        id
        email
        totalCount
        completedCount
      }
      clientMutationId
    }
  }  
  ",
  "variables": Object {
    "clientMutationId": "6.0",
    "complete": true,
    "id": "VG9kbzoyMjU=",
    "userId": "VXNlcjo5MA==",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 22`] = `
Object {
  "data": Object {
    "changeTodoStatus": Object {
      "clientMutationId": "6.0",
      "todo": Object {
        "complete": true,
        "id": "VG9kbzoyMjU=",
        "text": "Get A Customer 1",
      },
      "user": Object {
        "completedCount": 1,
        "email": "user@test.com",
        "id": "VXNlcjo5MA==",
        "totalCount": 1,
      },
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 23`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 24`] = `
Object {
  "query": "mutation clearCompletedTodos($userId: ID!, $clientMutationId: String) {
      clearCompletedTodos(input: {userId: $userId, clientMutationId: $clientMutationId}) {
        deletedTodoIds
        user {
          id
          completedCount
          totalCount
        }
        clientMutationId
      }
    }
    ",
  "variables": Object {
    "clientMutationId": "7",
    "userId": "VXNlcjo5MA==",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 25`] = `
Object {
  "data": Object {
    "clearCompletedTodos": Object {
      "clientMutationId": "7",
      "deletedTodoIds": Array [
        "VG9kbzoyMjU=",
      ],
      "user": Object {
        "completedCount": 0,
        "id": "VXNlcjo5MA==",
        "totalCount": 0,
      },
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 26`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 27`] = `
Object {
  "query": "mutation renameTodo($id: ID!, $text: String!, $clientMutationId: String) {
        renameTodo(
          input: { id: $id, text: $text, clientMutationId: $clientMutationId }
        ) {
          todo {
            id
            text
            complete
          }
          clientMutationId
        }
      }",
  "variables": Object {
    "clientMutationId": "9.0",
    "id": "VG9kbzoyMjU=",
    "text": "renamed 0",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 28`] = `
Object {
  "data": Object {
    "renameTodo": Object {
      "clientMutationId": "9.0",
      "todo": Object {
        "complete": false,
        "id": "VG9kbzoyMjU=",
        "text": "",
      },
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 29`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 30`] = `
Object {
  "query": "query User($email: String!, $status: Status, $first: Int, $after: String) {
        user(email: $email) {
          id
          email
          completedCount
          totalCount
          todos(status: $status, first: $first, after: $after) {
            edges {
              cursor
              node {
                id
                complete
                __typename
              }
            }
            pageInfo {
              endCursor
              hasNextPage
            }
          }
        }
      }",
  "variables": Object {
    "email": "user@test.com",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 31`] = `
Object {
  "data": Object {
    "user": Object {
      "completedCount": 0,
      "email": "user@test.com",
      "id": "VXNlcjo5MA==",
      "todos": Object {
        "edges": null,
        "pageInfo": Object {
          "endCursor": null,
          "hasNextPage": true,
        },
      },
      "totalCount": 0,
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 32`] = `
Object {
  "query": "mutation removeTodo($id: ID!, $userId: ID!, $clientMutationId: String) {
        removeTodo(
          input: { id: $id, userId: $userId, clientMutationId: $clientMutationId }
        ) {
          user {
            id
            email
            totalCount
            completedCount
          }
          clientMutationId
        }
      }",
  "variables": Object {
    "clientMutationId": "10.0",
    "id": "VG9kbzoyMjY=",
    "userId": "VXNlcjo5MQ==",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 33`] = `
Object {
  "data": Object {
    "removeTodo": Object {
      "clientMutationId": "10.0",
      "user": Object {
        "completedCount": 0,
        "email": "user@test.com",
        "id": "VXNlcjo5MQ==",
        "totalCount": 1,
      },
    },
  },
}
`;

exports[`Todos GraphQL Query & Mutation 34`] = `
"
=================================================="
`;

exports[`Todos GraphQL Query & Mutation 35`] = `
Object {
  "query": "query User($email: String!, $status: Status, $first: Int, $after: String) {
        user(email: $email) {
          id
          email
          completedCount
          totalCount
          todos(status: $status, first: $first, after: $after) {
            edges {
              cursor
              node {
                id
                complete
                __typename
              }
            }
            pageInfo {
              endCursor
              hasNextPage
            }
          }
        }
      }",
  "variables": Object {
    "email": "user@test.com",
  },
}
`;

exports[`Todos GraphQL Query & Mutation 36`] = `
Object {
  "data": Object {
    "user": Object {
      "completedCount": 0,
      "email": "user@test.com",
      "id": "VXNlcjo5MQ==",
      "todos": Object {
        "edges": null,
        "pageInfo": Object {
          "endCursor": null,
          "hasNextPage": true,
        },
      },
      "totalCount": 1,
    },
  },
}
`;
